test/runtests.jl

```diff
@@ -246,3 +246,28 @@ end
     @test isa(tuplegroups.A.i, Tuple)
     @test isa(tuplegroups.B.j, Tuple)
 end
+
+struct FreeParam{T,P<:NamedTuple} <: AbstractParam{T}
```

Comment:
If we're going to do this, then the macro to abstract away this boilerplate (and reduce chance of user error) should be included in this PR, I think.